### PR TITLE
(feat): adds e2e test with registration and login workflow

### PR DESCRIPTION
**Pull request by sjaghori**

### CODE REVIEW

**Comment by sjaghori** [comment]
https://github.com/codingcommons/tiny-tms/issues/10

File: e2e/specs/new-user-flow.spec.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+test.describe('registration process', { tag: ['@foo-bar'] }, () => {
+	const testEmail = 'foo@bar.com'
+	const testPassword = 'abc123abc123'
+
+	test('registers foo bar and logs into the app', async ({ page, baseURL }) => {
+		await page.goto(`${baseURL!}/signup`)
+
+		await test.step('sign up a new user', async () => {
+			const firstname = page.getByPlaceholder('Tiny')
```
Comment (benjaminstrasser) [inline e2e/specs/new-user-flow.spec.ts:11]: imo getByTestId is more stable. Also for all other instances.

File: src/routes/(auth)/signup/+page.server.ts [hunk 2]
```diff
@@ -40,6 +40,6 @@ export const actions = {
 			})
 		}
 
-		return message(form, 'Registration sucessful')
+		return message(form, 'Registration successful')
```
Comment (benjaminstrasser) [inline src/routes/(auth)/signup/+page.server.ts:43]: good catch! :rocket:

**Review by benjaminstrasser — COMMENTED** [review]
nice work!
just some small stuff.
The test is currently failing if you run it multiple times. See #10 
> Create a test/reset endpoint to reset the database, secured with an environment variable and a JWT.

What are thoughts on this approach?

File: e2e/specs/new-user-flow.spec.ts (hunk 1 above)
Comment (sjaghori) [reply e2e/specs/new-user-flow.spec.ts]: I totally agree ✅

**Comment by sjaghori** [comment]
> nice work! just some small stuff. The test is currently failing if you run it multiple times. See #10
> 
> > Create a test/reset endpoint to reset the database, secured with an environment variable and a JWT.
> 
> What are thoughts on this approach?

Agree we need a strategy for that, for now this setup is sufficient for the single test that we have. It's out of scope for this PR, there are some good ways (session storage, multiple workers etc.) that we can discuss.

Just by adding this simple registration/login flow test, we already protected ourself from non-obvious failing releases.

**Review by benjaminstrasser — APPROVED** [review]
LGTM!

**Comment by benjaminstrasser** [comment]
> > nice work! just some small stuff. The test is currently failing if you run it multiple times. See #10
> > > Create a test/reset endpoint to reset the database, secured with an environment variable and a JWT.
> > 
> > 
> > What are thoughts on this approach?
> 
> Agree we need a strategy for that, for now this setup is sufficient for the single test that we have. It's out of scope for this PR, there are some good ways (session storage, multiple workers etc.) that we can discuss.
> 
> Just by adding this simple registration/login flow test, we already protected ourself from non-obvious failing releases.

I already created a setup with basic playwright fixtures. PR coming soon!

**Comment by sjaghori** [comment]
@benjaminstrasser please make sure you add the data-testid's for the components in https://github.com/codingcommons/tiny-tms/pull/84

**Pull request merged**